core/capabilities/webapi/outgoing_connector_handler.go

```diff
@@ -11,28 +12,45 @@ import (
 	"github.com/smartcontractkit/chainlink-common/pkg/logger"
 	"github.com/smartcontractkit/chainlink/v2/core/services/gateway/api"
 	"github.com/smartcontractkit/chainlink/v2/core/services/gateway/connector"
+	"github.com/smartcontractkit/chainlink/v2/core/services/gateway/handlers/capabilities"
 	"github.com/smartcontractkit/chainlink/v2/core/services/gateway/handlers/common"
-	"github.com/smartcontractkit/chainlink/v2/core/services/gateway/handlers/webapicapabilities"
 )
 
-var _ connector.GatewayConnectorHandler = &ConnectorHandler{}
+var _ connector.GatewayConnectorHandler = &OutgoingConnectorHandler{}
 
-type ConnectorHandler struct {
+type OutgoingConnectorHandler struct {
 	gc            connector.GatewayConnector
+	method        string
 	lggr          logger.Logger
 	responseChs   map[string]chan *api.Message
 	responseChsMu sync.Mutex
 	rateLimiter   *common.RateLimiter
 }
 
-func NewConnectorHandler(gc connector.GatewayConnector, config Config, lgger logger.Logger) (*ConnectorHandler, error) {
+// Config is the configuration for OutgoingConnectorHandler.
+// Currently used by the WebApi Target and Compute Action capability & handler
+// TODO: handle retry configurations here CM-472
+// Note that workflow executions have their own internal timeouts and retries set by the user
+// that are separate from this configuration
+type Config struct {
+	IDGenerator func() string
+	RateLimiter common.RateLimiterConfig `toml:"rateLimiter"`
+}
```

Comment:
why are some dependencies put in the config and some are just passed to the constructor?

Reply:
🤔 good observation, it makes more sense to pass the IDGenerator as a parameter given is not truly a configuration that comes from a toml file like the RateLimiter. Will change it